sdk/core/core-client/src/serviceClient.ts

```diff
@@ -330,32 +318,21 @@ function flattenResponse(
           arrayResponse[key] = parsedHeaders[key];
         }
       }
-      return addResponse(arrayResponse);
+      return arrayResponse;
     }
 
     if (typeName === "Composite" || typeName === "Dictionary") {
-      return addResponse({
+      return {
         ...parsedHeaders,
         ...fullResponse.parsedBody
-      });
+      };
     }
   }
 
-  if (
-    bodyMapper ||
-    fullResponse.request.method === "HEAD" ||
-    isPrimitiveType(fullResponse.parsedBody)
-  ) {
-    return addResponse({
-      ...parsedHeaders,
-      body: fullResponse.parsedBody
```

Comment:
This is strange, since `HEAD` requests are explicitly supposed to not have a body from the HTTP spec. I hope no services are actually relying on sending body content for `HEAD` requests...

Reply:
I just realized why we were doing this, here: https://github.com/Azure/azure-sdk-for-js/blob/adeac3ecbfeeef1883106a8e7468cb911aa0ac76/sdk/core/core-client/src/deserializationPolicy.ts#L196. I think I should revert this back where the condition is for head only and return headers only in this case. Not sure if we need to include the boolean body, I will dig into the testserver and see how is that handled.